### PR TITLE
SNOW-199334 Revert SNOW-187770 and bring back 4 deleted public functions

### PR DESCRIPTION
**Pull request by sfc-gh-mrui**
To release SC 2.8.2 safer, AWS multiple-part upload API is disabled.
Bring back 4 deleted functions to avoid the possibility to break any
customers. 2 functions are in Utils. 2 functions are in Parameters.

### CODE REVIEW

File: ClusterTest/src/main/scala/net/snowflake/spark/snowflake/testsuite/LowMemoryStressSuite.scala [hunk 1]
```diff
@@ -76,16 +76,17 @@ class LowMemoryStressSuite extends ClusterTestSuiteBase {
...
+//    // Write to snowflake
+//    df.write
+//      .format(TestUtils.SNOWFLAKE_NAME)
+//      .options(TestUtils.sfOptionsNoTable)
+//      .option("dbtable", test_big_partition)
+//      .mode(SaveMode.Overwrite)
+//      .save()
+//
+//    log.info(
+//      s"""Finished the first multi-part upload test ======================""".stripMargin)
```
Comment (sfc-gh-bli) [inline ClusterTest/src/main/scala/net/snowflake/spark/snowflake/testsuite/LowMemoryStressSuite.scala:89]: can we remove this commented off code?

File: src/it/scala/net/snowflake/spark/snowflake/SnowflakeResultSetRDDSuite.scala [hunk 2]
```diff
@@ -1439,7 +1439,7 @@ class SnowflakeResultSetRDDSuite extends IntegrationSuiteBase {
     }
   }
 
-  test("test upload with AWS multiple parts upload API") {
+  ignore("test upload with AWS multiple parts upload API") {
```
Comment (sfc-gh-bli) [inline src/it/scala/net/snowflake/spark/snowflake/SnowflakeResultSetRDDSuite.scala:1442]: could you add comment here, to describe why we ignore this test and do we have plan to re-enable it?

File: src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala [hunk 3]
```diff
@@ -1372,46 +1372,66 @@ case class InternalS3Storage(param: MergedParameters,
...
+//            val usedTime = System.currentTimeMillis() - start
+//            CloudStorageOperations.log.info(
+//              s"""${SnowflakeResultSetRDD.WORKER_LOG_PREFIX}:
+//                 | Finish uploading file $file without AWS multiple parts API
+//                 | uploadTime=${Utils.getTimeString(usedTime)}
+//                 | compressedSize=${Utils.getSizeString(resultByteArray.length)}
+//                 |""".stripMargin.filter(_ >= ' '))
+//          }
+//        }
+//    }
```
Comment (sfc-gh-bli) [inline src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala:1434]: same here. if we don't need this code, can we remove it.

File: src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala [hunk 4]
```diff
@@ -1872,185 +1892,185 @@ object StorageUtils {
...
+//  }
+//
+//  override def close(): Unit = {
+//    // "dataSizeInBuffer is zero" means write() is never called.
+//    // For this case, the file upload is skipped.
+//    if (dataSizeInBuffer > 0) {
+//      uploadDataChunk(isLastChunk = true)
+//    }
+//  }
+//}
```
Comment (sfc-gh-bli) [inline src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala:2075]: same here

File: src/test/scala/net/snowflake/spark/snowflake/io/GcpSuite.scala [hunk 5]
```diff
@@ -43,29 +43,29 @@ class GcpSuite extends FunSuite {
...
+//    val dataInBuffer = s3UploadStream.getDataInBufferForTest()
+//    // Verify value to be correct
+//    for (i <- 1 to 128) {
+//      assert(dataInBuffer(i - 1) == i.toByte)
+//    }
+//    // write one more byte will cause exception because 's3Client' is not set
+//    assertThrows[Exception]({
+//      s3UploadStream.write(100)
+//    })
+//  }
```
Comment (sfc-gh-bli) [inline src/test/scala/net/snowflake/spark/snowflake/io/GcpSuite.scala:69]: same here

**Review by sfc-gh-bli — APPROVED** [review]
Please address my comments before merge.

File: ClusterTest/src/main/scala/net/snowflake/spark/snowflake/testsuite/LowMemoryStressSuite.scala (hunk 1 above)
Comment (sfc-gh-mrui) [reply ClusterTest/src/main/scala/net/snowflake/spark/snowflake/testsuite/LowMemoryStressSuite.scala:89]: I prefer to keep it.
The feature is disabled for SC 2.8.2 release.The feature will be re-enabled in the future. So, the test code needs to be brought back too. So it should be better to keep it. Does it make sense?

**Comment by sfc-gh-mrui** [comment]
> Please address my comments before merge.

All comments are addressed.

**Pull request merged**